### PR TITLE
Adding a check to find NPN-capable OpenSSL.

### PR DESCRIPTION
**Pull request by nicolasnoble**
Several comments:

-) We're always going to try building the embedded OpenSSL if we have it and if the system one isn't ALPN-capable.
-) I've removed the hardcoded test for the version number: this seems irrelevant now, especially since we have another check elsewhere that ensures the version of the submodule (which was the reason for that test to begin with).

### CODE REVIEW

File: test/build/openssl-npn.c [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+/* This is just a compilation test, to see if we have a version of OpenSSL with
+   NPN support installed. */
+
+#include <stdlib.h>
+#include <openssl/ssl.h>
+
+int main() {
+  SSL_get0_next_proto_negotiated(NULL, NULL, NULL);
+  return OPENSSL_NPN_UNSUPPORTED;
```
Comment (jboeuf) [inline test/build/openssl-npn.c]: This is slightly confusing but if you have a good rationale for it, that's fine. Maybe a comment...

**Comment by nicolasnoble** [comment]
I've done the 0.9 backport in jboeuf/grpc#1, targeting #2176.

File: test/build/openssl-npn.c (hunk 1 above)
Comment (nicolasnoble) [inline test/build/openssl-npn.c]: Updated.

Comment (jboeuf) [inline test/build/openssl-npn.c]: I don't see it. Did you push?

**Comment by jboeuf** [comment]
Just to make sure. Do you want me to add it to #2176?

**Comment by jboeuf** [comment]
travis is red for a makefile generation issue. I'll merge when it's green.

**Comment by nicolasnoble** [comment]
Crap, I didn't realize this still wasn't merged. And here I was, telling people around me this was done. I'm a bad person.

PR has been updated.

**Comment by nicolasnoble** [comment]
Also, @murgatroid99, have a quick look at the Makefile changes ?

**Comment by jboeuf** [comment]
happy to merge but travis python is red. @nathanielmanistaatgoogle do you think it's a flake?

**Comment by nicolasnoble** [comment]
Travis is all green now btw :-P

**Pull request merged**